File: backend/workiez/src/main/java/workiez/workiez/job/Job.java
package workiez.workiez.job;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import workiez.workiez.Application;
import workiez.workiez.user.District;
import workiez.workiez.user.Province;
import workiez.workiez.user.User;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Job {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long jobId;

    private String JobName;

    private String description;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Enumerated(EnumType.STRING)
    private District locationDistrict;

    @Enumerated(EnumType.STRING)
    private Province locationProvince;

    private String city;

    @Enumerated(EnumType.STRING)
    private JobStatus jobStatus;

    private String creationDateTime; //methna string kala dan dtabase ek del krla aaye run krla blnna

}

File: frontend/workiez/src/models/job.ts
import { District } from "./Enums/DistrictEnum"
import { JobStatus } from "./Enums/JobstatusEnum"
import { Province } from "./Enums/ProvinceEnum"
import { UserDTO } from "./UserDTO"
import { User } from "./user"

export interface Job {
    jobId: number,
    JobName: string,
    description: string
    user : User,
    locationDistrict: District,
    locationProvince:Province,
    city:string,
    jobStatus:JobStatus,
    creationDateTime:string
}